flytekit/remote/remote.py

```diff
@@ -1028,11 +1028,14 @@ def _get_image_names(entity: typing.Union[PythonAutoContainerTask, WorkflowBase]
             if isinstance(entity, WorkflowBase):
                 default_inputs = entity.python_interface.default_inputs_as_kwargs
 
+            from flytekit.configuration.plugin import get_plugin
```

Comment:
I include the import here as a workaround to the potential circular import issue (remote imports plugin, plugin imports remote)

Reply:
I could rename `additional_context` to `version_hash_additional_context`? I wouldn't rename it to `task_configs` because it does not necessarily contain the task configs, that only happens for a particular use case.

Moving it to a parameter of `register_script` seems sensible, but for the moment I'm thinking of keeping the PR as very easy to review/not changing too many methods. Will give it a try at a later point.